src/Compilers/CSharp/Test/Semantic/Semantics/WarningVersionTests.cs

```diff
@@ -139,13 +139,62 @@ public Program(int dummy)
     {
     }
 }";
-            CreateCompilation(source2, references: new MetadataReference[] { moduleReference }, options: TestOptions.ReleaseDll.WithWarningLevel(CodeAnalysis.Diagnostic.DefaultWarningLevel)).VerifyDiagnostics(
-                );
-            CreateCompilation(source2, references: new MetadataReference[] { moduleReference }, options: TestOptions.ReleaseDll.WithWarningLevel(5)).VerifyDiagnostics(
-                // (4,12): warning CS8822: Auto-implemented property 'Program.Property' must be fully assigned before control is returned to the caller.
+            var expectedIL = @"
+{
+  // Code size       14 (0xe)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  ldflda     ""Struct Program.<Property>k__BackingField""
+  IL_0006:  initobj    ""Struct""
+  IL_000c:  nop
+  IL_000d:  ret
+}
+";
+            // C# 10
+            var verifier = CompileAndVerify(
+                source2,
+                references: moduleReference,
+                options: TestOptions.DebugDll.WithWarningLevel(CodeAnalysis.Diagnostic.DefaultWarningLevel),
+                parseOptions: TestOptions.Regular10,
+                verify: Verification.Skipped);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("Program..ctor", expectedIL);
+
+            verifier = CompileAndVerify(
+                source2,
+                references: moduleReference,
+                options: TestOptions.DebugDll.WithWarningLevel(5),
+                parseOptions: TestOptions.Regular10,
+                verify: Verification.Skipped);
```

Comment:
Consider modifying those tests to also verify execution (`expectedOutput:` should show reads of auto-defaulted values)